File: examples/demo-typescript/tests/greet.test.ts
import { describe, it, expect } from "vitest";
import { hello, shout } from "../src/greet.js";

describe("greet", () => {
  it("hello returns a friendly greeting", () => {
    expect(hello("world")).toBe("hello, world");
  });

  it("shout uppercases hello", () => {
    expect(shout("world")).toBe("HELLO, WORLD");
  });
});

File: examples/demo-go/greet/greet_test.go
package greet

import "testing"

func TestHello(t *testing.T) {
	if got := Hello("world"); got != "hello, world" {
		t.Fatalf("Hello: got %q, want %q", got, "hello, world")
	}
}

func TestShoutUppercasesHello(t *testing.T) {
	if got := Shout("world"); got != "HELLO, WORLD" {
		t.Fatalf("Shout: got %q, want %q", got, "HELLO, WORLD")
	}
}
